specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,173 @@
+#### Types
+
+Before encoding, all `txn` parameters must be converted into primitive types:
+
+- `gh`: The genesis hash of the network, which is a base64 encoded string should be decoded. 
+- `snd`: The sender address should be decoded into it's public key bytes.
+- `rcv`: The receiver address should be decoded into it's public key bytes.
+- `xaid`: The asset ID should be converted into a uint64 integer.
+- `aamnt`: The amount to transfer should be converted into a uint64 integer.
+- `fee`: The fee for the transaction should be converted into a uint64 integer.
+- `fv`: The first valid round should be converted into a uint64 integer.
+- `lv`: The last valid round should be converted into a uint64 integer.
+
+#### Scheme
+
+Algorand uses [**msgpack**](https://www.npmjs.com/package/algo-msgpack-with-bigint) to encode the transaction before signing.
+
+
+#### Steps to sign
+
+- Decode the `txn` parameters as described above.
+- Sort fields alphabetically. 
+- Encode the sorted fields using msgpack.
+- Add a prefix `TX` in it's byte representation to the encoded message.
+- Sign the resulting byte array using the sender's private key with the `Ed25519` signature scheme.
+
+
+## Full `X-PAYMENT` header:
+
+```json
+{
+  "x402Version": 1,
+  "scheme": "exact",
+  "network": "algorand-mainnet",
+  "payload": {
+    "paymentGroup": [
+      {
+        "signature": "VxpW6emWwct7ymzyXa/F2T7rwepM71XwsslPKEdpVqmbcRUB28QMdYJuuLkeP4dnVoI7emmxT6Q/YktqHc8xBg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "fee": "2000",
+          "fv": "6631154",
+          "lv": "6632154",
+          "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "amnt": "0",
+          "type": "pay"
+        }
+      },
+      {
+        "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+          "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+          "type": "axfer",
+          "fee": "0",
+          "fv": "6631154",
+          "lv": "6632154",
+          "xaid": "168103"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification
+
+Steps to verify a payment for the `exact` scheme on Algorand:
+
+1. Verify the signature is valid for each transaction in the `paymentGroup`.
+  a. Encode the `txn` parameters as described in the encoding section.
+  b. Recover the public key of the sender from the `snd` or `asnd` field by decoding from the address.
+  c. Verify the signature against the encoded message using the recovered public key.
+2. Verify that the atomic group of transactions is valid and can be executed as a single unit.
```

Comment:
In the "`X-Payment` header payload" section above, it mentions the `paymentGroup` represents an atomic group. Does this mean it **MUST** be an atomic group? The examples don't appear to have a `grp` field, so I thought it might be optional, until now where it says to verify the atomic group can be executed together. Should we explain _how_ to do this? (e.g. hash of txids, although it's a little more complex than that)